File: AndroidViewModel/ViewModelDataBinding/app/src/main/java/com/haoyudu/test1/MainActivity.java
package com.haoyudu.test1;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;
import com.haoyudu.viewmodeldatabinding.R;
import com.haoyudu.viewmodeldatabinding.databinding.TestOuterActivityBinding;


public class MainActivity extends AppCompatActivity {
    ViewModelTest viewModelTest;
//    TextView textView;
//    Button btn;
    TestOuterActivityBinding binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding=DataBindingUtil.setContentView(this, R.layout.test_outer_activity);
        /**和主文件夹不在同一分级,无法访问到ViewModelTest.class*/
//        viewModelTest=ViewModelProviders.of(this).get(ViewModelTest.class);
        binding.setData(viewModelTest);
        binding.setLifecycleOwner(this);
    }
}

File: KotlinTest/KotlinNavigation/app/src/main/java/com/haoyudu/kotlinnavigationtest/MainActivity.kt
package com.haoyudu.kotlinnavigationtest

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.NavController
import androidx.navigation.fragment.findNavController
import androidx.navigation.ui.NavigationUI
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.icon_fragment_first.*
import kotlinx.android.synthetic.main.icon_fragment_second.*
import kotlinx.android.synthetic.main.icon_fragment_third.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //kotlin
        var navController: NavController =
            supportFragmentManager.findFragmentById(R.id.fragment)?.findNavController() as NavController
//        NavigationUI.setupActionBarWithNavController(this,navController)


        first_icon.setOnClickListener {
            navController.navigate(R.id.first_fragment)
        }
        second_icon.setOnClickListener {
            navController.navigate(R.id.second_fragment)
        }
        third_icon.setOnClickListener {
            navController.navigate(R.id.third_fragment)
        }

    }
}